core/Enclave/enclave_hsm.cpp

```diff
@@ -169,6 +169,15 @@ sgx_status_t enclave_create_key(ehsm_keyblob_t *cmk, size_t cmk_size)
         return SGX_ERROR_INVALID_PARAMETER;
     }
 
+    if ((cmk->metadata.keyusage == EH_KEYUSAGE_SIGN_VERIFY && cmk->metadata.keyspec == EH_AES_GCM_128) ||
+        (cmk->metadata.keyusage == EH_KEYUSAGE_SIGN_VERIFY && cmk->metadata.keyspec == EH_AES_GCM_192) ||
+        (cmk->metadata.keyusage == EH_KEYUSAGE_SIGN_VERIFY && cmk->metadata.keyspec == EH_AES_GCM_256) ||
```

Comment:
why not put this check into the switch...case dispatcher function, e.g,
ehsm_create_aes_key(cmk);
ehsm_create_sm4_key(cmk);

Reply:
ok, that's better